hub/api/dataset_handler.py

```diff
@@ -0,0 +1,180 @@
+from typing import Optional
+from hub.constants import DEFAULT_LOCAL_CACHE_SIZE, DEFAULT_MEMORY_CACHE_SIZE
+from .dataset import Dataset
+from hub.util.keys import dataset_exists
+
+
+class dataset:
+    def __new__(
+        cls,
+        path: str,
+        read_only: bool = False,
+        overwrite: bool = False,
+        public: Optional[bool] = True,
+        memory_cache_size: int = DEFAULT_MEMORY_CACHE_SIZE,
+        local_cache_size: int = DEFAULT_LOCAL_CACHE_SIZE,
+        creds: Optional[dict] = None,
+    ):
+        """Returns a Dataset object referencing either a new or existing dataset.
+
+        Args:
+            path (str): The full path to the dataset. Can be:-
+                - a Hub cloud path of the form hub://username/datasetname. To write to Hub cloud datasets, ensure that you are logged in to Hub (use 'activeloop login' from command line)
+                - an s3 path of the form s3://bucketname/path/to/dataset. Credentials are required in either the environment or passed to the creds argument.
+                - a local file system path of the form ./path/to/dataset or ~/path/to/dataset or path/to/dataset.
+                - a memory path of the form mem://path/to/dataset which doesn't save the dataset but keeps it in memory instead. Should be used only for testing as it does not persist.
+            read_only (bool): Opens dataset in read only mode if this is passed as True. Defaults to False.
+                Datasets stored on Hub cloud that your account does not have write access to will automatically open in read mode.
+            overwrite (bool): Overwrites the dataset if it already exists. Defaults to False.
+            public (bool, optional): Defines if the dataset will have public access. Applicable only if Hub cloud storage is used and a new Dataset is being created. Defaults to True.
+            memory_cache_size (int): The size of the memory cache to be used in MB.
+            local_cache_size (int): The size of the local filesystem cache to be used in MB.
+            creds (dict, optional): A dictionary containing credentials used to access the dataset at the path.
+                This takes precedence over credentials present in the environment. Currently only works with s3 paths.
+                It supports 'aws_access_key_id', 'aws_secret_access_key', 'aws_session_token', 'endpoint_url' and 'region' as keys.
+
+        Returns:
+            Dataset object created using the arguments provided.
+        """
+
+        if overwrite:
+            storage = get_storage_provider(path)
+            if dataset_exists(storage):
+                storage.clear()
+
+        return Dataset(
+            path=path,
+            read_only=read_only,
+            public=public,
+            memory_cache_size=memory_cache_size,
+            local_cache_size=local_cache_size,
+            creds=creds,
+        )
+
+    @staticmethod
+    def empty(
+        path: str,
```

Comment:
```suggestion
        uri: str,
```

can we replace all user-facing `path` references to `uri`? this is more descriptive and general

Reply:
@istranic said to keep path. all good here